app/controllers/account/utility_providers_controller.rb

```diff
@@ -0,0 +1,63 @@
+class Account::UtilityProvidersController < Account::AccountController
+  before_action :set_company, only: %i[new update]
+  before_action :set_billing_contract, only: :update
+
+  def index
+    @utility_providers = current_user.billing_contracts.page params[:page]
+  end
+
+  def search_company
+    @q = Company.joins(:account).ransack(params[:q])
+    @companies = @q.result.page(params[:page]).per(10)
+    render :companies
+  end
+
+  def new
+    if !current_user.companies.find_by(id: @company.id)
+      @billing_contract = BillingContract.new
+    else
+      flash[:warning] = 'Contract was associated!'
+      redirect_to %i[account utility_providers]
+    end
+  end
+
+  def update
+    if @billing_contract&.update(user_id: current_user.id)
+      flash[:success] = "Billing contract updated."
+      redirect_to %i[account utility_providers]
+    else
+      flash.now[:warning] = 'Wrong billing contract number!'
+      @billing_contract = BillingContract.new
```

Comment:
do you really need to render form with new billing contact? I think enough to render with instance in line 25, then form will be loaded with error messages

Reply:
If `@billing_contract` is nil I need create new. Changed to `@billing_contract ||= BillingContract.new`